src/pybind/mgr/dashboard/controllers/rgw.py

```diff
@@ -506,15 +506,32 @@ def set_quota(self, uid, quota_type, enabled, max_size_kb, max_objects, daemon_n
     def create_subuser(self, uid, subuser, access, key_type='s3',
                        generate_secret='true', access_key=None,
                        secret_key=None, daemon_name=None):
-        return self.proxy(daemon_name, 'PUT', 'user', {
-            'uid': uid,
-            'subuser': subuser,
-            'key-type': key_type,
-            'access': access,
-            'generate-secret': generate_secret,
-            'access-key': access_key,
-            'secret-key': secret_key
-        })
+        # pylint: disable=R1705
+        subusr_array = []
+        user = json.loads(self.get(uid, daemon_name))  # type: ignore
+        subusers = user["subusers"]
+        for sub_usr in subusers:
+            subusr_array.append(sub_usr["id"])
+        if subuser in subusr_array:
+            return self.proxy(daemon_name, 'POST', 'user', {
```

Comment:
@aaSharma14 could you briefly explain how the change fixed the issue? I tried to understand this one but couldn't. thanks.

Reply:
To update a subuser, we need to call the `POST` method, for creating a subuser, we need to call the `PUT` method. Previously we were calling `PUT` method for all the requests irrespective of creation or updation, thats why the stats were not getting updated. Now am checking if the subuser is there or not. If the subuser exists we will update it and call the POST method, if subuser does not exists we call the PUT method and carry on with the creation

Reply:
Okay, thanks